pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -103,43 +112,38 @@ func (c OpenShiftAPIServerOperator) sync() error {
 		return err
 	}
 
-	switch operatorConfig.Spec.ManagementState {
-	case operatorsv1.Managed:
-	case operatorsv1.Unmanaged:
-		return nil
-	case operatorsv1.Removed:
-		// TODO probably need to watch until the NS is really gone
-		if err := c.kubeClient.CoreV1().Namespaces().Delete(operatorclient.TargetNamespace, nil); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		return nil
-	default:
-		c.eventRecorder.Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfig.Spec.ManagementState)
+	if !management.IsOperatorManaged(operatorConfig.Spec.ManagementState) {
 		return nil
 	}
 
+	cond := operatorv1.OperatorCondition{
+		Type:   "PreRequirementNotReady",
```

Comment:
nit: prerequisite is probably better than prerequirement

Reply:
sure, fixed.